nodestream/pipeline/transformers/transformer.py

```diff
@@ -107,6 +111,9 @@ def drain_completed_tasks():
             for result in drain_completed_tasks():
```

Comment:
This looks like another contender for blocking the event loop. Should we make this also async friendly?

Reply:
Yup good catch, this is the same as the above 2 cases.